src/psbt.c

```diff
@@ -194,6 +194,22 @@ static const struct wally_tx_output *utxo_from_input(const struct wally_psbt *ps
                                      child_path, child_path_len); \
     }
 
+/* Add a taproot keypath to parent structs keypaths member */
+#define ADD_TAP_KEYPATH(PARENT) \
+    int PARENT ## _tap_keypath_add(struct PARENT *parent, \
+                               const unsigned char *pub_key, size_t pub_key_len, \
+                               const unsigned char *tapleaf_hashes, size_t tapleaf_hashes_len, \
+                               const unsigned char *fingerprint, size_t fingerprint_len, \
+                               const uint32_t *child_path, size_t child_path_len) { \
+        if (!parent) return WALLY_EINVAL; \
+        return wally_map_tap_keypath_add(&parent->taproot_leaf_hashes, \
+                                         &parent->taproot_leaf_paths, \
+                                         pub_key, pub_key_len, \
+                                         fingerprint, fingerprint_len, \
+                                         tapleaf_hashes, tapleaf_hashes_len, \
+                                         child_path, child_path_len); \
+    }
```

Comment:
As per other comment, I would not do this at the map interface level. You could have a local static function with this signature that this macro calls, but I suspect that would be just more code than its worth vs calling the 2 map functions separately.

Reply:
to be clear, do you mean I should modify the macro to call `wally_map_merkle_path_add` and `wally_map_keypath_add` since this `wally_map_tap_keypath_add` would be going away?

Reply:
done